src/python/pants/vcs/git.py

```diff
@@ -74,24 +74,31 @@ def changed_files(
         from_commit: str | None = None,
         include_untracked: bool = False,
         relative_to: PurePath | str | None = None,
+        omit_deleted_files: bool = False,
     ) -> set[str]:
         relative_to = PurePath(relative_to) if relative_to is not None else self.worktree
         rel_suffix = ["--", str(relative_to)]
+
+        def mk_cmd(diffspec: str) -> Iterable[str]:
+            cmd = ["diff", "--name-only"]
+            if omit_deleted_files:
+                cmd.append("--diff-filter=d")
+            cmd.append(diffspec)
+            cmd.extend(rel_suffix)
+            return cmd
+
         uncommitted_changes = self._git_binary._invoke_unsandboxed(
-            self._create_git_cmdline(
-                ["diff", "--name-only", "HEAD"] + rel_suffix,
-            )
+            self._create_git_cmdline(mk_cmd("HEAD"))
         )
 
         files = set(uncommitted_changes.splitlines())
         if from_commit:
             # Grab the diff from the merge-base to HEAD using ... syntax.  This ensures we have just
             # the changes that have occurred on the current branch.
-            committed_cmd = ["diff", "--name-only", from_commit + "...HEAD"] + rel_suffix
             committed_changes = self._git_binary._invoke_unsandboxed(
-                self._create_git_cmdline(committed_cmd)
+                self._create_git_cmdline(mk_cmd(from_commit + "...HEAD"))
             )
-            files.update(committed_changes.split())
```

Comment:
This should always have been splitlines() ...

Reply:
Do you want to file a different PR to fix the split vs splitlines issue? Or just throw it in this PR?

Reply:
Good call, will fix here.